xbmc/network/osx/darwinICMPPing.m

```diff
@@ -0,0 +1,221 @@
+/*
+ *      Copyright (C) 2015 Team MrMC
+ *      https://github.com/MrMC
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with MrMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <sysexits.h>
+#include <arpa/inet.h>
+
+#import "SimplePing.h"
+#import "darwinICMPPing.h"
+
+NSString * const kdidStartWithAddress         = @"didStartWithAddress";
+NSString * const kdidFailWithError            = @"didFailWithError";
+NSString * const kdidFailToSendPacket         = @"didFailToSendPacket";
+NSString * const kdidReceiveUnexpectedPacket  = @"didReceiveUnexpectedPacket";
+
+@interface SimplePingClient : NSObject<SimplePingDelegate>
++(void)pingHostname:(NSString*)hostName andResultCallback:(void(^)(NSString* result))result;
++(void)pingHostAddress:(NSData*)hostAddress andResultCallback:(void(^)(NSString* result))result;
+@end
+
+int darwinICMPPingHost(const char *host, unsigned int timeout_ms)
+{
+  // also checks if host is a name or ip address
+  struct sockaddr_in callAddress = {0};
+  callAddress.sin_len = sizeof(callAddress);
+  callAddress.sin_family = AF_INET;
+  callAddress.sin_addr.s_addr = inet_addr(host);
```

Comment:
Why not call darwinICMPPing(remoteIp, .. ) after lookup inet_addr and avoid duplicating a big chunk of code?